tests/scripts/verify-operator.sh

```diff
@@ -11,9 +11,42 @@ source ${SCRIPT_DIR}/.definitions.sh
 # Import the check definitions
 source ${SCRIPT_DIR}/checks.sh
 
-check_pod_ready "nvidia-driver-daemonset"
-check_pod_ready "nvidia-container-toolkit-daemonset"
-check_pod_ready "nvidia-device-plugin-daemonset"
-check_pod_ready "nvidia-dcgm-exporter"
-check_pod_ready "gpu-feature-discovery"
-check_pod_ready "nvidia-operator-validator"
+# wait for the nvidia-driver pod to be ready
+# If successful, then wait for the validator pod to be ready (this means that the rest of the pods are healthy)
+# collect log in case of failure
+start_time=$(date +%s)
+log_dir=${LOG_DIR}
+mkdir -p ${log_dir}
+while :; do
+    current_time=$(date +%s)
+    elapsed_time=$((current_time - start_time))
+    
+    # Check if total elapsed time is greater than exit
+    if [ $elapsed_time -gt $MAX_POD_STATUS_CHECK_TOTAL_TIME ]; then
+        echo "Total wait time exceeded ${MAX_POD_STATUS_CHECK_TOTAL_TIME} seconds. Exiting..."
+        cleanup_pod ${TEST_NAMESPACE} "gpu-operator"
+        exit 1
+    fi
+
+    check_pod_ready "nvidia-driver-daemonset" ${DAEMON_POD_STATUS_TIME_OUT} && \
+        check_pod_ready "nvidia-operator-validator" ${POD_STATUS_TIME_OUT}; status=$?
+    
+    if [ $status -ne 0 ]; then
+        not_ready_pod_status=$(kubectl get pods -n $TEST_NAMESPACE -o jsonpath="{.items[?(@.status.phase != 'Running')].metadata.name}")
+        if [ -n "$not_ready_pod_status" ]; then
+            for pod in $not_ready_pod_status; do
+                echo "Collecting logs for pod: $pod in dir ${log_dir}"
```

Comment:
As mentioned in a previous comment, lets extract this into a separate function, `collect_logs`, which we can define in a separate file.

Note, we have a script named `must-gather.sh` which we ask users to run when they are hitting operator issues. It collects a comprehensive set of logs to help us debug. Feel free to reuse some of the logic from there, specifically the logic to collect pod logs: https://github.com/NVIDIA/gpu-operator/blob/main/hack/must-gather.sh#L169-L196. If it makes sense, we can consider just running the `must-gather.sh` script directly from within our tests to collect logs.

Reply:
Also, I downloaded the logs generated from this failing pipeline, and all of the files are empty: https://github.com/shivakunv/gpu-driver-container/actions/runs/10425570121
Can you check this?

Reply:
I removed the while loop and added more delay since the gpu-operator sometimes takes longer to deploy. 
increased the daemon wait time to 15 minutes.

I kept the must-gather.sh script as it is, as it only requires the artifacts directory path for log collection. 
I have integrated it with the current code